Cabal/src/Distribution/Simple/BuildPaths.hs

```diff
@@ -67,6 +67,10 @@ srcPref distPref = distPref </> "src"
 hscolourPref :: HaddockTarget -> FilePath -> PackageDescription -> FilePath
 hscolourPref = haddockPref
 
+-- | Build info json file, generated in every build
+buildInfoPref :: FilePath -> FilePath
+buildInfoPref distPref = distPref </> "build-info.json"
```

Comment:
Possible inconsistency, file name is `build-info` but other flags are `buildinfo`. Any preferences?

Reply:
As far as I'm concerned buildinfo is a word.

Reply:
I choose consistency with `debug-info`, everything will be renamed to `build-info`